src/DeviceInfo.js

```diff
@@ -61,7 +64,7 @@ define(function(require) {
          * The device has touch events
          * @type {boolean}
          */
-        this.hasTouch = ('ontouchstart' in window);
+        this.hasTouch = ('ontouchstart' in window) || !!window.navigator.userAgent.match(ieTouchRegex);
```

Comment:
While I would have liked to get rid of DeviceInfo entirely, we still use it in some places that are not related to event handling, such as deciding the number of pages to render.  I think we want to treat Surfaces as "mobile" for now, so I added this UA check.  Surfaces do not have ontouchstart.  :rage: